src/metrics/mod.rs

```diff
@@ -74,13 +152,56 @@ impl Metrics {
         }
     }
 
-    pub fn delta(&self, snapshot: &mut Snapshot) -> u64 {
-        let curr = self.metric().value();
-        let prev = snapshot.prev.insert(*self, curr).unwrap_or(0);
+    pub fn delta(&self, snapshot: &mut MetricsSnapshot) -> u64 {
```

Comment:
Same question about the delta function here: can we replace callers in output/ which are doing `.delta(snapshot)` with `.counter().value()`?